python/cudf/cudf/_lib/table.pyx

```diff
@@ -71,106 +71,6 @@ cdef class Table:
-
-        cdef vector[unique_ptr[column]].iterator it = columns.begin()
-
-        # First construct the index, if any
-        cdef int i
-
-        index = None
-        if index_names is not None:
-            index_data = ColumnAccessor._create_unsafe(
-                {
-                    name: Column.from_unique_ptr(
-                        move(dereference(it + i))
-                    )
-                    for i, name in enumerate(index_names)
-                }
-            )
-            index = Table(data=index_data)
-
-        # Construct the data dict
-        cdef int n_index_columns = len(index_names) if index_names else 0
-        data = ColumnAccessor._create_unsafe(
-            {
-                name: Column.from_unique_ptr(
-                    move(dereference(it + i + n_index_columns))
-                )
-                for i, name in enumerate(column_names)
-            }
-        )
-
-        return Table(data=data, index=index)
-
-    @staticmethod
-    cdef Table from_table_view(
-        table_view tv,
-        object owner,
-        object column_names,
-        object index_names=None
-    ):
-        """
-        Given a ``cudf::table_view``, constructs a ``cudf.Table`` from it,
-        along with referencing an ``owner`` Python object that owns the memory
-        lifetime. If ``owner`` is a ``cudf.Table``, we reach inside of it and
-        reach inside of each ``cudf.Column`` to make the owner of each newly
-        created ``Buffer`` underneath the ``cudf.Column`` objects of the
-        created ``cudf.Table`` the respective ``Buffer`` from the relevant
-        ``cudf.Column`` of the ``owner`` ``cudf.Table``.
-        """
-        cdef size_type column_idx = 0
-        table_owner = isinstance(owner, Table)
-
-        # First construct the index, if any
-        index = None
-        if index_names is not None:
-            index_columns = []
-            for _ in index_names:
-                column_owner = owner
-                if table_owner:
-                    column_owner = owner._index._columns[column_idx]
-                index_columns.append(
-                    Column.from_column_view(
-                        tv.column(column_idx),
-                        column_owner
-                    )
-                )
-                column_idx += 1
-            index = Table(dict(zip(index_names, index_columns)))
-
-        # Construct the data dict
-        cdef size_type source_column_idx = 0
-        data_columns = []
-        for _ in column_names:
-            column_owner = owner
-            if table_owner:
-                column_owner = owner._columns[source_column_idx]
-            data_columns.append(
-                Column.from_column_view(tv.column(column_idx), column_owner)
-            )
-            column_idx += 1
-            source_column_idx += 1
-        data = dict(zip(column_names, data_columns))
-
-        return Table(data=data, index=index)
-
```

Comment:
Do we also want to get rid of the `view()` methods below to prevent the temptation to do `Table._from_date({'a': col}).view()`?

Reply:
Because now we have a bit of asymmetry in that logic for `Table -> view` lives in `Table` but the `view -> Table` logic lives in `utils.pyx`

Reply:
Yeah I agree that this asymmetry should be removed, I just didn't do it in this PR since it's already quite large. We also need to decide how we want to handle views with/without indexes if we remove it from the class: is the calling code responsible for calling `make_table_view` with the combined iterable of columns vs just the data columns, or do we make a new helper function? If you have an opinion on how to move forward we can squeeze that in here, otherwise toss that in a follow-up PR.

Reply:
@shwina since the need for a good common factory like `_from_data` is already evident in PRs like #8767 (see [here](https://github.com/rapidsai/cudf/pull/8767#discussion_r677878805) in particular), I'd vote to push the move of the `Table.[mutable_][data_]view` methods to a subsequent PR. It should definitely still be done, but perhaps not so critical yet.

Reply:
That's fine with me!